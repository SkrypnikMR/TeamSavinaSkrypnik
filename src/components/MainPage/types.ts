export type TMainPage = {
    getSockJSConnection: () => void;
    disconnect: () => void;
}
