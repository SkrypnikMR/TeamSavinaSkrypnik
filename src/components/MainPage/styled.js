import styled from "styled-components";

export const StMainPage = styled.div`
    background: #404040;
`;