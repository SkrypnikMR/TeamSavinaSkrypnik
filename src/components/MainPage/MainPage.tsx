import React from 'react';

import Header from '../Header';
import MainContent from '../MainContent';
import Footer from '../Footer';
import { useTheme } from '../Hook/useTheme';

import { StMainPage } from './styled';

const MainPage = () => {
    const { colors, theme } = useTheme();
    return (
        <StMainPage colors={colors} theme={theme}>
            <Header />
            <MainContent />
            <Footer/>
        </StMainPage>
        
    );
};

export default MainPage;
