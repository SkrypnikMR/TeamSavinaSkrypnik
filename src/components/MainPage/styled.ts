import styled from 'styled-components';
import { MAIN_PAGE_BG } from '../UI/baseLayout';

export const StMainPage = styled.div`
    background: ${MAIN_PAGE_BG}
`;
