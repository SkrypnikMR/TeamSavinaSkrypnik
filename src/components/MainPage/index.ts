import { connect } from 'react-redux';
import { Dispatch } from 'redux';
import { getSockJSConnection } from '../../store/game/actions';
import MainPage from './MainPage';

const mapDispatchToProps = (dispatch: Dispatch) => ({
    getSockJSConnection: () => dispatch(getSockJSConnection()),
});

export default connect(null, mapDispatchToProps)(MainPage);
