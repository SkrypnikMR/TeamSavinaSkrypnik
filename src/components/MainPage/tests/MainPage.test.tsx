import React from 'react';
import MainPage from '../MainPage';
import { shallowSmart, mountSmart } from '../../../../__tests__/testHelper';


describe('RoomsContainer', () => {
    it('Should match snapshot', () => {
        const component = shallowSmart(<MainPage />);
        expect(component.html()).toMatchSnapshot();
    });
    it('should render StMainPage', () => {
        const component = mountSmart(<MainPage/>);
        expect(component.find('styled__StMainPage')).toHaveLength(1);
    });
    it('should render Header', () => {
        const component = mountSmart(<MainPage/>);
        expect(component.find('Header')).toHaveLength(1);
    });
    it('should render MainContent', () => {
        const component = mountSmart(<MainPage/>);
        expect(component.find('MainContent')).toHaveLength(1);
    });
        it('should render Footer', () => {
        const component = mountSmart(<MainPage/>);
        expect(component.find('Footer')).toHaveLength(1);
    });
});
