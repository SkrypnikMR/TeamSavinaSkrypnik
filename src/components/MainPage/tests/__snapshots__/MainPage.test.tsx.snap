// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage Should match snapshot 1`] = `"<div class=\\"styled__StMainPage-sc-jna7hk-0 hvcDzu\\"><div class=\\"styled__StHeader-sc-1d7yllk-0 ibTsld\\"><div class=\\"styled__StLogo-sc-1lhyjdk-0 eFYqgc\\"><img src=\\"public/assets/images/game-development.png\\"/><div><button type=\\"button\\" font-size=\\"20px\\" class=\\"styled__StButton-sc-1y35zax-0 klRECi\\">game_zone</button><button type=\\"button\\" font-size=\\"20px\\" class=\\"styled__StButton-sc-1y35zax-0 WqRKw\\">statistics</button></div></div><div class=\\"styled__StControl-sc-1l7fu2q-0 bcijKV\\"> <button id=\\"theme_btn\\" type=\\"button\\" value=\\"dark\\" color=\\"white\\" font-size=\\"26px\\" width=\\"40px\\" class=\\"styled__StButton-sc-1y35zax-0 dxIorm\\">ðŸ’¡</button><button id=\\"en\\" type=\\"button\\" value=\\"en\\" color=\\"white\\" font-size=\\"26px\\" width=\\"40px\\" class=\\"styled__StButton-sc-1y35zax-0 hfKumd\\">english_translate_btn</button><button id=\\"ru\\" type=\\"button\\" value=\\"ru\\" color=\\"white\\" font-size=\\"26px\\" width=\\"40px\\" class=\\"styled__StButton-sc-1y35zax-0 edZCPS\\">russian_translate_btn</button><img src=\\"public/assets/images/log-out.png\\"/></div></div><div class=\\"styled__StMainContent-sc-1j4ztaa-0 bVTJbg\\"><div class=\\"styled__StRoomsContainer-sc-2jyga5-0 iHEkTO\\"><div class=\\"styled__StAddRoom-sc-5w938c-0 djgSFI\\">+</div></div><div class=\\"styled__StGameContent-sc-rfvg7b-0 cPjyTE\\"><div class=\\"styled__StHello-sc-3v9sfb-0 bwTzmh\\"><img src=\\"https://i.gifer.com/origin/51/518532f622d962be53e2e1f8989263a8_w200.gif\\"/><p>hi, Unknown!!!<br/>joing_room</p></div></div></div><div class=\\"styled__StFooter-sc-1xbiugi-0 bClKzk\\">create_forÂ <a href=\\"https://deveducation.com/\\" target=\\"_blank\\" rel=\\"noreferrer\\">dev{Education}</a></div></div>"`;
